ignite/metrics/precision.py

```diff
@@ -23,9 +30,12 @@ def update(self, output):
         indices = torch.max(y_pred, 1)[1]
         correct = torch.eq(indices, y)
         pred_onehot = to_onehot(indices, num_classes)
-        correct_onehot = to_onehot(indices[correct], num_classes)
         all_positives = pred_onehot.sum(dim=0)
-        true_positives = correct_onehot.sum(dim=0)
+        if correct.sum() == 0:
+            true_positives = torch.zeros(num_classes)
```

Comment:
This can fail later at 
```
  File "/usr/local/lib/python3.5/dist-packages/ignite/metrics/precision.py", line 44, in update
    self._true_positives += true_positives
```
if user uses CUDA
The error is 
```
TypeError: add_ received an invalid combination of arguments - got (torch.cuda.FloatTensor), but expected one of:
 * (float value)
      didn't match because some of the arguments have invalid types: (torch.cuda.FloatTensor)
 * (torch.FloatTensor other)
      didn't match because some of the arguments have invalid types: (torch.cuda.FloatTensor)
 * (torch.SparseFloatTensor other)
      didn't match because some of the arguments have invalid types: (torch.cuda.FloatTensor)
 * (float value, torch.FloatTensor other)
 * (float value, torch.SparseFloatTensor other)
```

Reply:
Maybe, we could use `true_positives = torch.zeros_like(all_positives)`